x/mongo/driver/session/client_session.go

```diff
@@ -266,11 +262,23 @@ func (c *Client) UpdateRecoveryToken(response bson.Raw) {
 	c.RecoveryToken = token.Document()
 }
 
-// ClearPinnedServer sets the PinnedServer to nil.
-func (c *Client) ClearPinnedServer() {
-	if c != nil {
-		c.PinnedServer = nil
+// ClearPinnedResources clears the pinned server and/or connection associated with the session.
```

Comment:
No, it's just simpler to do so because it avoids additional special-casing and checking for load-balanced mode in different places. If pinning to a server was resource-intensive, conditional checks would make sense, but we're just keeping a reference that later gets set to nil, so I opted to leave it as-is.

Reply:
That makes sense, thank you for the explanation.